crates/uv/tests/it/pip_compile.rs

```diff
@@ -12661,3 +12661,33 @@ fn prune_unreachable() -> Result<()> {
 
     Ok(())
 }
+
+/// Allow resolving a package that requires a Python version that is not available, as long as it
+/// includes static metadata.
+///
+/// See: <https://github.com/astral-sh/uv/issues/8767>
+#[test]
+fn unsupported_requires_python_static_metadata() -> Result<()> {
+    let context = TestContext::new("3.11");
+    let requirements_in = context.temp_dir.child("requirements.in");
+    requirements_in.write_str("interpreters-pep-734 <= 0.4.1 ; python_version >= '3.13'")?;
+
+    uv_snapshot!(context
+        .pip_compile()
+        .arg("--universal")
```

Comment:
Should we add a `--no-build` for extra checks?

Reply:
It actually doesn't work with `--no-build` (we fully fail, since we don't even consider sdists). Unsure whether it should... I could go either way on it.

Reply:
That is a tough one: Should you be able to resolve with `--no-build`, when we already know that we can't install with `--no-build`?

Can you open an to track that separately?

Reply:
Yeah.